### ISSUE TITLE
Secrets

### ISSUE THREAD

**Issue by erikh** [tinyci/tinyci#6]
Secrets could be implemented as such:

`tinycli` could be used to register documents with a name for a given repository. The user must have "admin" privileges on the github repository as a basic gating mechanism that we already know with the tools we have in the API already. Once admin pages land in the UI, it would also live there.

In tinyci's control plane API, one could manipulate a secret from the control plane by name, or by map of filename -> document. The former would be used for administrative needs; the latter for runners. These maps would be provided by datasvc. queuesvc and uisvc can expose them to the runner and user, respectively.

The runners themselves could do this in an independent way since the API doesn't prescribe how it's served to the user, but for now we're going to target via the filesystem, so runners might populate a directory with the contents of this map; similar to how kubernetes secrets work.

Additionally, an AES key will be provided for the repository's secrets, and will be needed to decrypt the secrets on disk. User flow looks something like: `cat file | aes decrypt > secrets!`.

Finally, some attempt needs to be made to obscure logs in the face of secrets being sprayed to them. While far from perfect, this can keep google or whoever from indexing valuable data for future searches, but little else I think.

What might be more useful is a tinyCI capability that only allows some users to read logs. I will add a separate ticket for this.
